### PR TITLE
Use Run#getLogReader() with built-in encoding management when accessing run logs

### PR DESCRIPTION
**Pull request by ndeloof**
Prefer Reader over InputStream to ensure configured charset is used.
@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/model/Run.java [hunk 1]
```diff
@@ -1910,7 +1911,7 @@ public synchronized void save() throws IOException {
      */
     @Deprecated
     public @Nonnull String getLog() throws IOException {
-        return Util.loadFile(getLogFile(),getCharset());
+        return IOUtils.toString(getLogReader());
```
Comment (andresrc) [inline core/src/main/java/hudson/model/Run.java]: :ant: `IOUtils.toString` does not close the stream.

Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Run.java]: +1

**Comment by daniel-beck** [comment]
@ndeloof This doesn't actually deprecate anything…?

**Comment by oleg-nenashev** [comment]
The code itself looks to be reasonable, but the PR title is strange

**Comment by oleg-nenashev** [comment]
@ndeloof is the new title better?

**Comment by oleg-nenashev** [comment]
@ndeloof merge conflict :(

**Comment by jglick** [comment]
Was this fixing a particular filed bug? Seems like it should have been testable.

**Pull request closed without merging**